### DISCUSSION TITLE
How can I get the bound size of a control without creating a new one?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11831] [Q&A] [answered]
I am creating a resizable control base on textbox, just like this:
![2](https://github.com/AvaloniaUI/Avalonia/assets/30720036/89363e2e-57f4-4433-8443-c1edfae61bd0)

Now I want to predict the bounds. Height and bounds. Width of the textbox after scaling according to the new fontsize. If the bounds. Height and bounds.width exceeds the value I specified, then it will not be scaled. But to calculate this, I had to create a new textbox. 

Or scaling the existing textbox to a new fontsize. If the calculated value does not match after zooming, the old value will be restored, but this will cause the control to flicker, which is very bad.

How can I calculate the bound size without specifically creating a new textbox?

**Comment by timunie** [comment]
file a minimal sample of what you have so far. RN it's not easy for me to understand the desired outcome. Normally I'd say just define MaxHeight, MaxWidth but not sure if it will work for you.

**Reply by kyuranger** [reply, marked as answer]
MaxWidth doesn't work when scaling it by fontsize. Currently I use another way by wrapping a Viewbox outside it:
```
<Viewbox Stretch="Uniform">
  <Textblock></Textblock>
</Viewbox
```
And now I can scale it by width and height. Problem solved!
